### PR TITLE
Issue 6967

### PR DESCRIPTION
**Pull request by MadWasp79**
If its normal implementation will refactor for favorites and tracks tabs

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/myplaces/FavoritesActivity.java [hunk 1]
```diff
@@ -187,13 +193,39 @@ private void setTabs(List<TabItem> mTabs) {
 
 	@Override
 	public void onAttachFragment(Fragment fragment) {
-		fragList.add(new WeakReference<>(fragment));
+		if (fragment instanceof FavoritesFragmentStateHolder) {
+			if (intentParams != null && intentParams.getInt(TAB_ID, -1) != -1) {
+				Bundle b = new Bundle();
+				int tabId = intentParams.getInt(TAB_ID, FAV_TAB);
+				if (tabId == FAV_TAB && fragment instanceof FavoritesTreeFragment) {
+					b.putString(FavoritesFragmentStateHolder.GROUP_NAME_TO_SHOW,
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/myplaces/FavoritesActivity.java]: Use OOP. 
all the logic of putInt, getInt should be hidden behind FavoritesFragmentStateHolder.

NotesFragment belongs to Plugin so it couldn't be used here !!!

File: OsmAnd/src/net/osmand/plus/myplaces/FavoritesActivity.java [hunk 2]
```diff
@@ -24,8 +24,11 @@
 import net.osmand.plus.OsmandSettings;
 import net.osmand.plus.R;
 import net.osmand.plus.activities.FavoritesTreeFragment;
+import net.osmand.plus.activities.MapActivity;
 import net.osmand.plus.activities.TabActivity;
+import net.osmand.plus.audionotes.NotesFragment;
 import net.osmand.plus.helpers.ImportHelper;
+import net.osmand.plus.osmedit.OsmEditsFragment;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/myplaces/FavoritesActivity.java]: NotesFragment belongs to Plugin so it couldn't be used here !!!

File: OsmAnd/src/net/osmand/plus/myplaces/FavoritesActivity.java [hunk 3]
```diff
@@ -24,8 +24,11 @@
 import net.osmand.plus.OsmandSettings;
 import net.osmand.plus.R;
 import net.osmand.plus.activities.FavoritesTreeFragment;
+import net.osmand.plus.activities.MapActivity;
 import net.osmand.plus.activities.TabActivity;
+import net.osmand.plus.audionotes.NotesFragment;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/myplaces/FavoritesActivity.java]: OsmEditsFragment belongs to Plugin so it couldn't be used here !!!

File: OsmAnd/src/net/osmand/plus/myplaces/FavoritesActivity.java (hunk 1 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/myplaces/FavoritesActivity.java]: OK, think I get it. Will do.

**Comment by vshcherb** [comment]
Fix #6967

**Pull request merged**